stats/src/main/java/com/facebook/stats/mx/JVMStatsExporter.java

```diff
@@ -71,9 +104,47 @@ public JVMStatsExporter(
     String... beanNamePatterns) 
       throws JMException {
     this.stats = stats;
-    this.statNamePattern = Pattern.compile(statNamePattern);
+    Pattern namePattern = Pattern.compile(statNamePattern);
```

Comment:
Instead of removing the local member variable, just replace it with a `statNameBuilder` member variable. This will allow you to maintain compatibility with the existing API, e.g.:

    private final StatsNameBuilder statsNameBuilder;
    …
    public JVMStatsExporter(
      Stats stats,
      String statNamePattern,
      String... beanNamePatterns
    ) throws JMException {
      this(stats, patternFilter(Pattern.compile(statNamePattern)), beanNamePatterns);
    }
    …
    private static StatsNameBuilder patternFilter(Pattern pattern) {
      return (bean, attribute, key) -> {
        String name = getStatName(bean, attribute, key);
   
        return pattern.matcher(name).matches() ? Optional.of(name) : Optional.empty();
      };
    }